helm-charts/splunk-otel-collector/templates/config/_otel-k8s-cluster-receiver-config.tpl

```diff
@@ -198,11 +203,16 @@ service:
   telemetry:
     metrics:
       address: 0.0.0.0:8889
-  {{- if eq (include "splunk-otel-collector.distribution" .) "eks/fargate" }}
-  extensions: [health_check, memory_ballast, k8s_observer]
-  {{- else }}
-  extensions: [health_check, memory_ballast]
-  {{- end }}
+  extensions:
+    - health_check
+    - memory_ballast
+    {{- if eq (include "splunk-otel-collector.distribution" .) "eks/fargate" }}
```

Comment:
This sounds unrelated to this PR.

Reply:
Kind of. 
But if I had kept it at it is, it would've been a bit complete, as multiple blocks should be added I guess.
If we just split out that `extensions` array, it provides greater flexibility to add/remove more extensions in the future.